src/models/CourtDeployer.js

```diff
@@ -0,0 +1,300 @@
+    const { court } = this.previousDeploy
+    const Court = await this.environment.getArtifact('Court')
+
+    if (court && court.address) await this._loadCourt(Court, court.address)
+    else await this._deployCourt(Court)
+  }
+
+  async loadOrDeployRegistry() {
+    const { registry } = this.previousDeploy
+    const JurorsRegistry = await this.environment.getArtifact('JurorsRegistry')
+
+    if (registry && registry.address) await this._loadRegistry(JurorsRegistry, registry.address)
+    else await this._deployRegistry(JurorsRegistry)
+  }
+
+  async loadOrDeployVoting() {
+    const { voting } = this.previousDeploy
+    const Voting = await this.environment.getArtifact('CRVoting')
+
+    if (voting && voting.address) await this._loadVoting(Voting, voting.address)
+    else await this._deployVoting(Voting)
+  }
+
+  async loadOrDeployTreasury() {
+    const { treasury } = this.previousDeploy
+    const Treasury = await this.environment.getArtifact('CourtTreasury')
+
+    if (treasury && treasury.address) await this._loadTreasury(Treasury, treasury.address)
+    else await this._deployTreasury(Treasury)
+  }
+
+  async loadOrDeploySubscriptions() {
+    const { subscriptions } = this.previousDeploy
+    const Subscriptions = await this.environment.getArtifact('CourtSubscriptions')
+
+    if (subscriptions && subscriptions.address) await this._loadSubscriptions(Subscriptions, subscriptions.address)
+    else await this._deploySubscriptions(Subscriptions)
+  }
+
+  async setModules() {
+    const sender = await this.environment.getSender()
+    const modulesGovernor = await this.controller.getModulesGovernor()
+
+    if (modulesGovernor === sender) {
+      logger.info('Setting modules...')
+      const { controller } = this.config
+      const ids = [controller.court, controller.treasury, controller.voting, controller.registry, controller.subscriptions]
+      const implementations = [this.court, this.treasury, this.voting, this.registry, this.subscriptions].map(i => i.address)
+      await this.controller.setModules(ids, implementations)
+      logger.success('Modules set successfully')
+    } else {
+      logger.warn('Cannot set modules since sender is no longer the modules governor')
+    }
+  }
+
+  async transferGovernor() {
+    const sender = await this.environment.getSender()
+    const modulesGovernor = await this.controller.getModulesGovernor()
+    const { governor: { modules } } = this.config
+
+    if (modulesGovernor === sender) {
+      logger.info(`Transferring modules governor to ${modules} ...`)
+      await this.controller.changeModulesGovernor(modules)
+      logger.success('Modules governor transferred successfully')
+    } else if (modulesGovernor === modules) {
+      logger.success(`Modules governor is already set to ${modules}`)
+    } else {
+      logger.warn('Modules governor is already set to another address')
+    }
+  }
+
+  /** loading methods **/
+
+  async _loadController(Controller, address) {
```

Comment:
This really got me, my brain definitely thought `Controller` was a type and not a variable 🙈